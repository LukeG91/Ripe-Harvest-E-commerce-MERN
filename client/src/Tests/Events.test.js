/* Importing the libraries and file that I need */
import React from "react";
import renderer from "react-test-renderer";
import EventsPage from "../components/events/Events";

/* Creating a snapshot test to check that the Events page renders correctly */
it("I am running a snapshot test in order to ensure that the Events page component renders correctly.", () => {
  const tree = renderer.create(<EventsPage />).toJSON();
  expect(tree).toMatchSnapshot();
});

/* Resource used:
     ===============
     YouTube video:
     Title of video: Learn React Snapshot Testing with Example
     Date published: March 17, 2020
     Published by: RakibTG
     Link to video: https://www.youtube.com/watch?v=wfFw05TaBfM&t=475s
     =================================================================
*/
