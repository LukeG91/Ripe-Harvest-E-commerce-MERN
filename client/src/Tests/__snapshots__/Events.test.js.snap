// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`I am running a snapshot test in order to ensure that the Events page component renders correctly. 1`] = `
<div>
  <h1
    className="EventsPageHeading"
  >
    Upcoming Events:
  </h1>
  <div
    className="eventsCarousel"
  >
    <div>
      <div
        style={
          Object {
            "textAlign": "center",
          }
        }
      >
        <div
          style={
            Object {
              "textAlign": "center",
              "width": "100%",
            }
          }
        >
          <div
            className="box"
            style={
              Object {
                "marginBottom": "20px",
                "marginTop": "20px",
                "textAlign": "center",
              }
            }
          >
            <div
              style={
                Object {
                  "maxHeight": "600px",
                  "maxWidth": "100%",
                }
              }
            >
              <div
                onMouseDown={[Function]}
                onTouchEnd={[Function]}
                onTouchStart={[Function]}
              >
                <div
                  className="carousel-container"
                  style={
                    Object {
                      "backgroundColor": "darkgrey",
                      "borderRadius": "10px",
                      "height": "600px",
                      "maxWidth": "100%",
                    }
                  }
                >
                  <div
                    className="carousel-item fade"
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchStart={[Function]}
                    style={
                      Object {
                        "maxHeight": "600px",
                        "maxWidth": "100%",
                      }
                    }
                  >
                    <img
                      alt="Oranjezicht night market Saturday the 28th of August: 6pm - 10pm"
                      className="carousel-image"
                      src="slider1.jpg"
                      style={
                        Object {
                          "borderRadius": "10px",
                          "objectFit": "cover",
                        }
                      }
                    />
                    <div
                      className="carousel-caption-bottom"
                      style={
                        Object {
                          "fontSize": "1.5em",
                          "fontWeight": "bold",
                        }
                      }
                    >
                      Oranjezicht night market Saturday the 28th of August: 6pm - 10pm
                    </div>
                  </div>
                  <div
                    className="carousel-item fade"
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchStart={[Function]}
                    style={
                      Object {
                        "maxHeight": "600px",
                        "maxWidth": "100%",
                      }
                    }
                  >
                    <img
                      alt="V&A Waterfront fresh produce market Saturday the 4th of September: 10am - 6pm"
                      className="carousel-image"
                      src="slider6.jpg"
                      style={
                        Object {
                          "borderRadius": "10px",
                          "objectFit": "cover",
                        }
                      }
                    />
                    <div
                      className="carousel-caption-bottom"
                      style={
                        Object {
                          "fontSize": "1.5em",
                          "fontWeight": "bold",
                        }
                      }
                    >
                      V&A Waterfront fresh produce market Saturday the 4th of September: 10am - 6pm
                    </div>
                  </div>
                  <div
                    className="carousel-item fade"
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchStart={[Function]}
                    style={
                      Object {
                        "maxHeight": "600px",
                        "maxWidth": "100%",
                      }
                    }
                  >
                    <img
                      alt="Old biscuit mill Saturday the 11th of September: 10am - 10pm"
                      className="carousel-image"
                      src="slider5.jpg"
                      style={
                        Object {
                          "borderRadius": "10px",
                          "objectFit": "cover",
                        }
                      }
                    />
                    <div
                      className="carousel-caption-bottom"
                      style={
                        Object {
                          "fontSize": "1.5em",
                          "fontWeight": "bold",
                        }
                      }
                    >
                      Old biscuit mill Saturday the 11th of September: 10am - 10pm
                    </div>
                  </div>
                  <a
                    className="prev"
                    onClick={[Function]}
                  >
                    ❮
                  </a>
                  <a
                    className="next"
                    onClick={[Function]}
                  >
                    ❯
                  </a>
                  <div
                    className="dots"
                  >
                    <span
                      className="dot"
                      onClick={[Function]}
                    />
                    <span
                      className="dot"
                      onClick={[Function]}
                    />
                    <span
                      className="dot"
                      onClick={[Function]}
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="eventsFooterContainer"
  >
    <div
      className="footerParentDiv"
    >
      <div
        className="footerMainDiv"
      >
        <p>
          Ripe Harvest || Copyright 2021
        </p>
      </div>
    </div>
  </div>
</div>
`;
